code/WorkInProgress/DrMelonsStuff.dm

```diff
@@ -196,17 +203,7 @@
 		if (!src.occupant)
 			boutput("<span class='alert'>There's no one inside!</span>")
 			return
-		src.occupant.visible_message("<span class='notice'>[src.occupant] gets out of the bath.</span>", "<span class='notice'>You get out of the bath.</span>")
-		src.occupant.pixel_y = 0
-		src.occupant.layer = initial(occupant.layer)
-		src.occupant.set_loc(get_turf(src))
-		src.vis_contents -= src.occupant
-		src.occupant = null
-		src.UpdateOverlays(null, "fluid_overlay")
-		src.UpdateOverlays(null, "bath_edge")
-		src.on_reagent_change()
-		if (src.reagents.total_volume)
-			playsound(src.loc, "sound/misc/splash_1.ogg", 70, 3)
+		Exited(src.occupant, user.loc)
```

Comment:
You shouldn't be explicitly calling `Exited()` -- set_loc will do it for you

Reply:
thank you, I misunderstood how Exited was used; these should all be fixed to use `set_loc`